src/ecommerce.interfaces.ts

```diff
@@ -96,28 +96,69 @@ export interface SDKECommerceAPI extends IECommerceShared {
         customFlags?: SDKEventCustomFlags,
         eventOptions?: SDKEventOptions
     ): void;
+    setCurrencyCode(code: string): void;
+
+    /*
+     * @deprecated
+     */
     logPurchase(
         transactionAttributes: TransactionAttributes,
         product: SDKProduct | SDKProduct[],
         clearCart?: boolean,
         attrs?: SDKEventAttrs,
         customFlags?: SDKEventCustomFlags
     ): void;
+    /*
+     * @deprecated
+     */
     logRefund(
         transactionAttributes: TransactionAttributes,
         product: SDKProduct | SDKProduct[],
         clearCart?: boolean,
         attrs?: SDKEventAttrs,
         customFlags?: SDKEventCustomFlags
     ): void;
-    setCurrencyCode(code: string): void;
+}
+
+interface ExtractedActionAttributes {
+    "Affiliation"?: string;
+    "Coupon Code"?: string;
+    "Total Amount"?: number;
+    "Shipping Amount"?: string;
+    "Tax Amount"?: string;
+    "Checkout Option"?: string;
+    "Checkout Step"?: number;
+    "Transaction ID"?: string;
+}
+interface ExtractedProductAttributes {
+    "Coupon Code"?: string;
+    "Brand"?: string;
+    "Category"?: string;
+    "Name"?: string;
+    "Id"?: string;
+    "Item Price"?: string;
+    "Quantity"?: string;
```

Comment:
a customer is allowed to add it as a string or number, but then we parse it into a number https://github.com/mParticle/mparticle-web-sdk/blob/master/src/ecommerce.js#L282-L287